form.js

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable require-jsdoc */
+const formName = document.querySelector('#name');
+const email = document.querySelector('#email');
+const formtext = document.querySelector('#text');
+const inputData = {};
+const storage = window.localStorage;
+function storageAvailable(type) {
+  let storage;
+  try {
+    storage = window[type];
+    const x = '__storage_test__';
+    storage.setItem(x, x);
+    storage.removeItem(x);
+    return true;
+  } catch (e) {
+    return e instanceof DOMException && (
+      e.code === 22 ||
+      e.code === 1014 ||
+      e.name === 'QuotaExceededError' ||
+      e.name === 'ERROR_QUOTA_REACHED') && (storage && storage.length !== 0);
+  }
+}
+
+function retrieveFormData() {
+  if (storageAvailable('localStorage')) {
+    const formDataString = storage.getItem('formData');
+    const formData = JSON.parse(formDataString);
+    return formData;
+  }
+  return false;
+}
+
+function populateForm() {
+  const formData = retrieveFormData();
+  if (formData) {
+    if (formData.name) {
+      formName.value = formData.name;
+    }
+    if (formData.email) {
+      email.value = formData.email;
+    }
+    if (formData.text) {
+      formtext.value = formData.text;
+    }
+  }
+}
+populateForm();
+
+formName.addEventListener('change', () => {
+  inputData.name = formName.value;
+  inputData.email = email.value;
+  inputData.text = formtext.value;
+  storage.setItem('formData', JSON.stringify(inputData));
+});
+
+email.addEventListener('change', () => {
+  inputData.name = formName.value;
+  inputData.email = email.value;
+  inputData.text = formtext.value;
+  storage.setItem('formData', JSON.stringify(inputData));
+});
+
+formtext.addEventListener('change', () => {
+  inputData.name = formName.value;
+  inputData.email = email.value;
+  inputData.text = formtext.value;
+  storage.setItem('formData', JSON.stringify(inputData));
+});
```

Comment:
- [ ] Kindly make sure that when the user loads the page, the code checks if there are any saved data in the local storage, and if so the input fields are pre-filled with the data. 
( _When the user refreshes the browser the data in the form should be persistent and to make it so you can follow the code below_ )

```
window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};
```